### ISSUE TITLE
[Kubermatic-Installer] [GO] Boilerplate

### ISSUE THREAD

**Issue by pkavajin** [kubermatic/kubermatic#1872]
To get the backend development started, we'll need some boilerplate here.
Should be about building the minimal blocks needed to further be able to incrementally build features.

**Acceptance criteria**
* There should be an "install" Command  in the kubermatic-installer binary.
* A mandatory parameter should be a manifest file.
* The manifest file (JSON) should be read and parsed into business structures.
* The installer should compare the version specified in the manifest with its own, if it differs, it should abort and write an error statement.
* Define types/interfaces for tasks so we can later easily execute and test all individual installation steps.
    * Tasks should be able to save data in some kind of context for other tasks to use
    * Tasks should be able to retrieve data from previous tasks
* Define a pipeline for the tasks to get executed, keep in mind that there are dependant tasks and some which we can run in parallel.

**Comment by kron4eg** [comment]
I think this is already done, isn't it @xrstf?

**Comment by xrstf** [comment]
yep
